### PR TITLE
SAK-48959 Announcements support removal of data when site is hard deleted

### PR DESCRIPTION
**Pull request by jkjanetschek**
…leted

### CODE REVIEW

**Comment by ern** [comment]
julian I have a change on the way to improve the hard delete fundamentals, if you have more of these assign the PR's to me and keep them as drafts, thanks.

**Comment by ern** [comment]
I would be better if you put all of your changes into a branch and share the branch with me ;)

**Comment by jkjanetschek** [comment]
I put all my draft implementations for a sakai life Cylce in a 23.x branch: [https://github.com/jkjanetschek/sakai/tree/23.x_liveCycle_draft](url)

This includes the following tools

- Calendar: BaseCalendarService implements HardDeleteAware
- Rubrics: RubricsServiceImpl implements HardDeleteAware
- Samigo: AssessmentEntityProducer implements HardDeleteAware
- Chat2: ChatEntityProducer implements HardDeleteAware
- Resources: BaseContentService implements HardDeleteAware
- Lessons: LessonsBuilderEntityProducer implements HardDeleteAware
- LTI: BasicLTISecurityServiceImpl implements HardDeleteAware
- Mailarchive: BaseMailArchiveService implements ContextOberver 
- Messageforum: DiscussionForumServiceImpl implements HardDeleteAware
- Poll: PollListManagerImpl implements HardDeleteAware
- RWiki: RwikiObjectServiceImpl implements HardDeleteAware
- SignUp: SignupEntityProducer implements HardDeleteAware
- PostEm: new class GradeBookService  implements SiteRemovalAdvisor
- Syllabus: SyllabusServiceImpl implements HardDeleteAware
- Hierarchy (delegated access): hierarchyServiceImpl implements SiteRemovalAdvisor

File: announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/BaseAnnouncementService.java [hunk 1]
```diff
@@ -121,7 +123,11 @@
  */
 @Slf4j
 public abstract class BaseAnnouncementService extends BaseMessage implements AnnouncementService, ContextObserver,
-		EntityTransferrer, ContentExistsAware {
+		EntityTransferrer, ContentExistsAware, SiteRemovalAdvisor
+{
+	/** private constants definitions */
+	private final static String SAKAI_ANNOUNCEMENT_TOOL_ID = "sakai.announcements";
+	private static final String PORTLET_CONFIG_PARM_MERGED_CHANNELS = "mergedAnnouncementChannels";
```
Comment (ern) [inline announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/BaseAnnouncementService.java:130]:
```suggestion
```

**Pull request merged**